### PR TITLE
fix(client|redux|react): orders area fixes

### PR DESCRIPTION
**Pull request by boliveira**
## Description

- This introduces a number of fixes/refactorings for the `orders` area:
  - Changed `getGuestOrders` client to drop support
  for the legacy endpoint.
  - Added missing `fetchGuestOrders` action.
  - Removed `getGuestOrder` client as it is replaced by
  the `getOrder` client.
  - Added `getGuestOrderLegacy` and a corresponding action
   to add support for the retrieval of guest orders using
   the legacy endpoint to ease migration.
  - Refactored orders reducer to be easier to understand.
  - Renamed `getOrders` client to `getUserOrders` and
  its corresponding action `fetchOrders` to
  `fetchUserOrders` to improve consistency.
- Changed clients which were still defined to use the
`legacy` endpoint to use the new endpoints.
- Fixed a lot of incorrect types.
- Renamed some subscriptions clients and actions.
- Renamed actions `fetchListing` to `fetchProductListing` and `fetchSet` to
`fetchProductSet`.
- Renamed `getUserCredit` client to `getUserCredits`.

BREAKING CHANGE: The following exports were renamed:
- OrderStatus -> MerchantOrderStatus
- deleteRecipientFromTopic -> deleteSubscriptionTopicRecipient
- getOrders -> getUserOrders
- getUserCredit (client) -> getUserCredits
- clearAllUnsubscribeRecipientFromTopic -> clearAllUnsubscribeSubscriptionTopicRecipientRequests
- clearUnsubscribeRecipientFromTopic -> clearUnsubscribeSubscriptionTopicRecipientRequest
- fetchListing -> fetchProductListing
- fetchListingFactory -> fetchProductListingFactory
- fetchOrders -> fetchUserOrders
- fetchOrdersFactory -> fetchUserOrdersFactory
- fetchSet -> fetchProductSet
- fetchSetFactory -> fetchProductSetFactory
- fetchUserCredit -> fetchUserCredits
- getMerchantsFromOrder -> getOrderMerchants
- getReturnOptionsFromOrder -> getOrderReturnOptions
- getUserCredit (selector) -> getUserCredits
- unsubscribeFromSubscription -> unsubscribeSubscription
- unsubscribeFromSubscriptionFactory -> unsubscribeSubscriptionFactory
- unsubscribeRecipientFromTopic -> unsubscribeSubscriptionTopicRecipient
- unsubscribeRecipientFromTopicFactory -> unsubscribeSubscriptionTopicRecipientFactory

## Checklist

- [X] The commit message follows our guidelines
- [X] Tests for the respective changes have been added
- [X] The code is commented, particularly in hard-to-understand areas
- [X] The labels and/or milestones were added

## Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

**Review by jhairabreu — CHANGES_REQUESTED** [review]
TS issues on the client package were recently addressed, so i ask you to double check if all of these changes didn't cause any new TS issues (in the client package). Please check it out before merging this. thanks! 🙂

File: packages/client/src/orders/types/getUserOrders.types.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import type { Config } from '../../types';
+import type { Orders } from '../../orders/types';
+import type { User } from '../../users/authentication/types/user.types';
+
+export type GetUserOrders = (
+  id: User['id'],
+  query?: GetUserOrdersQuery,
+  config?: Config,
+) => Promise<Orders>;
```
[the remark this answers is not in the document]
Comment (boliveira) [reply packages/client/src/orders/types/getUserOrders.types.ts:9]: Yes, I have missed that. Just updated, can you check, please? Thank you!

File: packages/client/src/orders/__tests__/getUserOrders.test.ts [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
 
-describe('getOrders', () => {
+describe('getUserOrders', () => {
   const spy = jest.spyOn(client, 'get');
 
   it('should handle a client request successfully', async () => {
     mswServer.use(fixtures.success(mockOrdersResponse));
 
-    await expect(getOrders(userId)).resolves.toEqual(mockOrdersResponse);
+    await expect(getUserOrders(userId)).resolves.toEqual(mockOrdersResponse);
```
[the remark this answers is not in the document]
Comment (boliveira) [reply packages/client/src/orders/__tests__/getUserOrders.test.ts:17]: Fixed the mocks to be compliant with the types expected by this test. Can you take a look at it, please? Thank you!

File: packages/client/src/orders/getGuestOrderLegacy.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { adaptError } from '../helpers/client/formatError';
```
[the remark this answers is not in the document]
Comment (boliveira) [reply packages/client/src/orders/getGuestOrderLegacy.ts:1]: Added a test for this client which I missed, thank you for alerting me! Can you check, please?

**Review by jhairabreu — APPROVED** [review]

**Review by giuliamml — APPROVED** [review]

**Pull request merged**